### ISSUE TITLE
SAMBA issues surfaced in FD811 with indirect WoofCE implications

### ISSUE THREAD

**Issue by CollaboratorGCM** [puppylinux-woof-CE/woof-CE#1876]
**Edit: This has been updated. It is a matter of implementation differences that is not apparent.**

I am not sure how to approach this problem. It may not be a bug in the packages, but, a problem/bug in implementation.

I do NOT want to take anything away when any developer takes steps for safety. But, if the step causes inconsistencies, then a consensus may be needed to resolve. 

That consensus can take many forms such that all PUPs and DOGs agree and makes public/private a common solution so that the majority of the userbase can help each other when the need arises. It could be in a common document or a common implementation or somewhere in the middle or ... so that PUPs and DOGs alike can easily talk to each other for common SAMBA sharing resources.

Tell me if I am wrong in describing this problem

### Scenario
Within days of each other, the GA release of FossaPUP arrives as well as v811 of FATDOG.

I setup each distro to serve a folder to LAN users. On each distro; namely FossaPUP-v95 and FATDOG-v811 a folder named Download on each distro they each, separately, will serve for LAN user access.
Traditionally, I mount resources as root user and share folder owned by root. These 2 actions have served well until I attempted to do the same with the WoofCE PUPs in an attempt to access the folders on FATDOG where I started to unsurface a couple anomalies. Next, I noticed that it wasn't just WoofCEs, but also other distros accessing FD811, too. Looking thru the smb.conf, I did not see anything that would hint at a clue, but tried everything from using smb.conf from FD5 & FD6 & LH &,,, nothing was clicking. So I setup the latest FD810 on one PC and FD811 on another to discover the same problems seen from the others...even as I used the very same smb.conf on each. Curiously, I tried to access the resource as user spot, mounting from FD811 to FD810's resource. **BAM!** Worked, AND, IT WORKS FROM FD810 to FD811 using the same mount command(s). AND using the same mount commands from 2020 PUPs also are now working to the FD8+ servers. As well as MS access requests with changes that are slightly different from requests to PUP servers.

So we have issues with differences in implementations across versions in FD8+ where there are some inconsistencies in how it and PUPs handle server services; namely the matter of root NOT be an available SAMBA user in FD8+, yet is perfectly normal in WoofCE PUPs.

Any wonder if this can confuse users.

### Questions
What is the best approach for both the PUP and DOG developers for Simple Samba Server? And could something consistent be beneficial to all parties? Or should the departing directions be published somewhere evident so that Windows and Linux users alike would see, if departing implementations is the future?

**This is a question for best approach for understanding and client-server usage on LANs with PUPs, DOGs, MACs and Windows.**

P.S. Is there a dialogue that should surface for best practices if something is known by one set of developers that the others should know but were not aware? i did Google around especially on the latest SAMBA version-releases and came up with nothing.

If anyone wants to see, I can share my console log of my tests findings from the various PUP to DOG runs (and vice-versa). Below is a starting log. The kernel versions for FD811 and FD810 are 5.4.60 and 4.19.92
```
#echo "Test to determine if local access is successful"
Test to determine if local access is successful
# mount -o password=woofwoof //localhost/Downloads /mnt/Test
mount error(13): Permission denied
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs)
# 
# echo "Test smb.conf for errs"
Test smb.conf for errs
# testparm
Load smb config files from /etc/samba/smb.conf
rlimit_max: increasing rlimit_max (1024) to minimum Windows limit (16384)
Processing section "[Downloads]"
Processing section "[Media]"
Processing section "[printers]"
Loaded services file OK.
Server role: ROLE_STANDALONE

Press enter to see a dump of your service definitions

# Global parameters
[global]
	map to guest = Bad Password
	printcap name = cups
	security = USER
	server string = Puppy Samba Server
	workgroup = WORKGROUP
	idmap config * : backend = tdb


[Downloads]
	path = /mnt/sda2/Downloads
	read only = No


[Media]
	path = /mnt/sda2/Music
	read only = No


[printers]
	browseable = No
	comment = All Printers
	guest ok = Yes
	path = /var/spool/samba
	printable = Yes
#
```

**Comment by jamesbond3142** [comment]
@CollaboratorGCM, I don't really get what you're trying to report here. Is this a Fatdog problem? Is this a Puppy problem? Are you expecting Puppy to behave more like Fatdog? Are you expecting Fatdog to behave more like Puppy? 

In all cases, this is a Puppy development site. Everything that you report here has to be Puppy-centric. If you want to suggest a new feature for Puppy (based on what you like from Fatdog or from any other distro for that matter), that's fine. If you want to suggest that Puppy should behave like Fatdog (or any other distro for that matter), that's fine, too. 

The opposite, however, is NOT OKAY. Nobody in the Fatdog team reads this site (apart from curiosity reasons) and they will NOT RESPOND to any issues you open here, as it is highly inappropriate. Puppy developers will similarly ignore anything other than Puppy stuff. 

In addition, when it comes to Samba, please be specific on what exactly are you doing. Samba is a complex matter and has tons of configuration options. I'm lost between you switching OS in every sentence and mixing configuration. The smb.conf you posted at the end of your post above, is neither from Puppy nor from Fatdog (but it's more similar to the one from Puppy). It is best, for testing and explaning, to use the default smb.conf; because once you change it, it becomes a Samba problem (and not Puppy or Fatdog problem anymore) and when that happens, it should be posted in the Forum.

**Comment by CollaboratorGCM** [comment]
Correct expression. And thanks for reviewing this post. As noted it is not a bug-bug, but more of a difference in understanding between the PUPs and current DOG.

I thought to post on the FD GIT, but the answer you share would also apply there as well.

Here's a summary of this problem experience, then I will explain what I am suggesting.
For years, I (and maybe others too) have added increasing LAN units, most of which have smb ability built-in. Over those years for many PUPs and FDs distros, the root user is the default for the shipped PUPs in SSM setups to make it easy for use to get going to share something from their PCs. There were even documents produced on use of SSM and how LAN users would access shared resources.

For those that want to access that from other PUPs & DOGs the mount command is the same. For MACs and Windows, I (others may have too) create a root user with the same password as is used on PUPs and they, too, can share media/data allowing access in the same way as the PUPs and FDs.

So this means that everyone sharing is using the exact same methods for local LAN use. 

This means that all accesses follow the same "mount" protocols, no matter what LAN unit that is sharing. Windows mounts to shares using rootuser, MACS mount using rootusers, TV with SMB clients built in are setup to mount-access shares using rootuser, **everyone** mounts any of the shares using the same rootuser.

BUT FD8+ is designed to not allow that rootuser with no "stated" circumvention.

**This post is designed to ALERT the developers of this in case they have not noticed this problem.**

In my case, FD811 and FossaPUP95 are both released in close proximity to each other. So I stumble onto something not seen or used by me, before. And since I am looking to make permanent these distro's use, I happened to have stumbled onto this incompatibility in their mount needs with no way to bring them into alignment so that all other LAN units can continue to use the same methods of access for all of their access needs no matter who on the LAN is sharing.

This now presents an unforeseen exception that affects all LAN users who would need to get to what FD811 is sharing. Its is inconsistent in how it traditionally is done.

I could merely abandon FD811 for this sharing, but that does not address the larger issue. Since PUPs for years have used the SSM as a simple effectively safe manner of sharing, maybe FD8+ knows something that PUPs should know. If so, is that knowledge begging a need for change? 

Or is there an oversight that is existing that needs discussion of some sort with FD that they may not have considered where the combined thoughts could come to an agreement so that SSM PUP users for past years can either continue. 

Right now, 
mount -o password=common-password  //LANunit/resource  /mnt/resource
only works to access PUPs, WINs, MACs, Tablets, Phones which are sharing....But not for accessing anything from FDs.

FD is not bad, it just doesn't allow past PUPs to use their old method that a homeowner has been employing for ease of use in their home and among their family.

My married daughter who has 3 kids has 31 devices on her home LAN from local camera storage to homework files for school and university to music from everyone to pics among everyone to video among everyone to doorbell and it keeps growing. So in their case we implemented an common sharing function that is a single simple user/password for all their home sharing requirements. So for example to introduce something that doesn't conform, traditionally, introduces a level of complexity that I just ran into. If it just me, that might be OK, but for 5 people and work related visitors, etc; one can see the compounding of this complexity.

### Suggestion
Maybe a simple dialogue of advantage-disadvantage since we're all in the same kennel. If both sides agree, then maybe a common approach is best. If there is consistency, there is no need for any exception documentation. If it remains inconsistent, then it should be taken to the forum so they know it will impact.

**Comment by CollaboratorGCM** [comment]
This response is long overdue. AND, I APOLOGIZE! I have been a little under the weather but am now OK. Nothing serious.

Since posting this, I had a conversatiion where a couple of suggestions were presented. Reviewing those and then retracing my steps, I was convinced that I had done all the same things as have been done since SAMBA became a standard part of the Puppy offering years ago.

So I whip out the same document I had completed with development years ago and retraced the steps.

I have discovered that I "may" have left out a key step. Further, I was absolutely sure I had done that step weeks ago. BUT, reviewing the console (terminal) histories of the multiple FDs I had tested, I may not have. (I have set up this so many times in my past, I could not imagine I have overlooked something important.)

This mis-step on my part led me to "think" that the FATDOG implementation was inconsistent.

I WAS WRONG! I AM SO SORRY FOR RAISING THIS ISSUE.

As it stand, at this time, there is NO inconsistency when the steps are followed in that old SAMBA user guide.

I will be making a new SAMBA guide for FATDOG (FD) users. FD offers a different utility for SAMBA sharing. So I will make a separate document using the FD tools.

The current guide for Puppy distros continues to be correct and consistent in setup and use.

The process is the same, no matter which PUP/DOG is used. And the client mounts are the same regardless of PUP/DOG, assuming, the "setup" steps in the guide(s) are followed.

AGAIN, I APOLOGIZE!
